dev/bots/prepare_package.dart

```diff
@@ -20,72 +20,220 @@ const String MINGIT_FOR_WINDOWS_URL = 'https://storage.googleapis.com/flutter_in
 
 /// Error class for when a process fails to run, so we can catch
 /// it and provide something more readable than a stack trace.
-class ProcessFailedException extends Error {
-  ProcessFailedException([this.message, this.exitCode]);
+class ProcessRunnerException implements Exception {
+  ProcessRunnerException(this.message, [this.result]);
 
-  String message = '';
-  int exitCode = 0;
+  final String message;
+  final ProcessResult result;
+  int get exitCode => result.exitCode ?? -1;
 
   @override
-  String toString() => message;
+  String toString() {
+    String output = 'ArchivePublisherException';
+    if (message != null) {
+      output += ': $message';
+    }
+    final String stderr = result?.stderr ?? '';
+    if (stderr.isNotEmpty) {
+      output += ':\n${result.stderr}';
+    }
+    return output;
+  }
+}
+
+enum Branch { dev, beta, release }
+
+String getBranchName(Branch branch) {
+  switch (branch) {
+    case Branch.beta:
+      return 'beta';
+    case Branch.dev:
+      return 'dev';
+    case Branch.release:
+      return 'release';
+  }
+  return null;
+}
+
+Branch fromBranchName(String name) {
+  switch (name) {
+    case 'beta':
+      return Branch.beta;
+    case 'dev':
+      return Branch.dev;
+    case 'release':
+      return Branch.release;
+    default:
+      throw new ArgumentError('Invalid branch name.');
+  }
+}
+
+/// A helper class for classes that want to run a process, have the stderr and
+/// stdout reported as the process runs, and capture the stdout.
+class ProcessRunner {
+  ProcessRunner({
+    ProcessManager processManager,
+    this.subprocessOutput: true,
```

Comment:
It's not clear from the name what this does -- can you document it or rename it to be more clear?